extensions/samples/electron-webpack/src/AuthProvider.ts

```diff
@@ -0,0 +1,162 @@
+                },
+                system: {
+                    loggerOptions: {
+                        loggerCallback(loglevel, message, containsPii) {
+                            console.log(message);
+                        },
+                        piiLoggingEnabled: false,
+                        logLevel: LogLevel.Info,
+                    },
+                },
+            });
+            return authProvider;
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async login(): Promise<AccountInfo> {
+        const tokenRequest: SilentFlowRequest = {
+            scopes: [],
+            account: null,
+        };
+        const authResponse = await this.getToken(tokenRequest);
+        return this.handleResponse(authResponse);
+    }
+
+    async getToken(
+        tokenRequest: SilentFlowRequest
+    ): Promise<AuthenticationResult> {
+        try {
+            let authResponse: AuthenticationResult;
+            const account = this.account || (await this.getAccount());
+            if (account) {
+                tokenRequest.account = account;
+                return (authResponse = await this.getTokenSilent(tokenRequest));
+            } else {
+                return (authResponse = await this.getTokenInteractive(
+                    tokenRequest
+                ));
+            }
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async getTokenSilent(
+        tokenRequest: SilentFlowRequest
+    ): Promise<AuthenticationResult> {
+        try {
+            return await this.clientApplication.acquireTokenSilent(
+                tokenRequest
+            );
+        } catch (error) {
+            console.log(
+                "Silent token acquisition failed, acquiring token using pop up"
+            );
+            return await this.getTokenInteractive(tokenRequest);
+        }
+    }
+
+    async getTokenInteractive(
+        tokenRequest: SilentFlowRequest
+    ): Promise<AuthenticationResult> {
+        try {
+            const openBrowser = async (url: any) => {
+                await shell.openExternal(url);
+            };
+
+            const interactiveRequest: InteractiveRequest = {
+                ...tokenRequest,
+                openBrowser,
+                successTemplate:
+                    "<h1>Successfully signed in!</h1> <p>You can close this window now.</p>",
+                errorTemplate:
+                    "<h1>Oops! Something went wrong</h1> <p>Check the console for more information.</p>",
+            };
+            return this.clientApplication.acquireTokenInteractive(
+                interactiveRequest
+            );
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    private async getAccount(): Promise<AccountInfo> {
+        const cache = this.clientApplication.getTokenCache();
+        const currentAccounts = await cache.getAllAccounts();
+
+        if (currentAccounts === null) {
```

Comment:
Does getAllAccounts always returns an array? Could you check for !currentAccounts.length here instead?

Reply:
Good catch. Updated!